src/FakeItEasy/Creation/IProxyOptions.cs

```diff
@@ -14,5 +14,7 @@ internal interface IProxyOptions
         IEnumerable<Action<object>> ProxyConfigurationActions { get; }
 
         IEnumerable<Expression<Func<Attribute>>> Attributes { get; }
+
+        string Name { get; }

```

Comment:
I'm not too happy with this, because `Name` isn't really an option of the proxy... However the alternative would have been quite painful, requiring the addition of new parameters in zillions of places. And strictly speaking, `ProxyConfigurationActions` is also not really an option of the proxy (i.e. it's not used for creating the proxy with Castle Core). Maybe we should just rename `IProxyOptions` to `IFakeCreationOptions`?